### PR TITLE
swaysettings: init at 0.3.0

### PR DESCRIPTION
**Pull request by aacebedo**
###### Description of changes
Added  the swaysettings tool.
The tool is great to edit some part of the configuration of a desktop environment based on sway.
homepage is here https://github.com/ErikReider/SwaySettings 

###### Things done
- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Closing this because of poor, below novice, quality.

**Comment by aacebedo** [comment]
Nice comment. Very helpful

**Comment by AndersonTorres** [comment]
OK, let's restart this.

File: pkgs/tools/system/swaysettings/default.nix [hunk 1]
```diff
@@ -0,0 +1,75 @@
+{
+  pkgs,
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:2]:
```suggestion
{ lib
, stdenv
```

No one puts the whole `pkgs` on the scope. Instead, the packages eventually needed are listed on the input set.

File: pkgs/tools/system/swaysettings/default.nix [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  meson,
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:3]: Commas come before the arguments, nor after them.

```suggestion
  , meson
```

File: pkgs/tools/system/swaysettings/default.nix [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  libhandy,
+  json-glib,
+  libgee,
+  pantheon,
+  libxml2,
+  accountsservice,
+  gobject-introspection,
+  dbus
+}:
+with pkgs;
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:24]:
```suggestion
```

As said before, `pkgs` are not put under the whole scope.

File: pkgs/tools/system/swaysettings/default.nix [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+pkgs.stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:25]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/system/swaysettings/default.nix [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  pname = "swaysettings";
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:26]: Are you using Notepad or whatever? Indent the code.

File: pkgs/tools/system/swaysettings/default.nix [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  version = "0.0";
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:27]: If the software you are bringing to Nixpkgs has no formal release, use something like `version = "unstable-YYYY-MM-DD";`

File: pkgs/tools/system/swaysettings/default.nix [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  src = pkgs.fetchFromGitHub {
+    owner = "ErikReider";
+    repo = "SwaySettings";
+    rev = "master";
+    hash = "sha256-EVGY4LbiH+RB56sYGOkYCdOc01LqbJIg9mJjgvFd4oY=";
+  };
+
+  nativeBuildInputs = [
+                        meson
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:36]:
```suggestion
  meson
```

File: pkgs/tools/system/swaysettings/default.nix [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+                  ];
+
+  meta = with pkgs.lib; {
+    description = "A GUI for configuring your sway desktop";
+    longDescription = ''
+      Sway settings enables configuring parts of a sway desktop environment
+      such as application or icons themes.
+    '';
+    homepage = "https://github.com/ErikReider/SwaySettings";
+    license = licenses.gpl3;
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:67]:
```suggestion
    license = licenses.gpl3Plus;
```

`gpl3` is deprecated. Use `gpl3Plus` or `gpl3Only` accordingly.

File: pkgs/tools/system/swaysettings/default.nix [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    platforms = platforms.linux;
+  };
+
+  postPatch = ''
+    chmod +x /build/source/build-aux/meson/postinstall.py # not necessary here
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:72]: If not needed, remove it.

Also, `meta` should be the last thing on the expression.

File: pkgs/top-level/all-packages.nix [hunk 10]
```diff
@@ -35435,4 +35435,6 @@ with pkgs;
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
 
   mictray = callPackage ../tools/audio/mictray { };
+
+  swaysettings = callPackage  ../tools/system/swaysettings { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:35439]:
```suggestion
  swaysettings = callPackage  ../tools/system/swaysettings { };
```

This is not a system tool.
This is an application, it should be in `../applications/misc`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Have you ever read the manual?

https://nixos.org/manual/nixpkgs/stable/

At least the quickstart:

https://nixos.org/manual/nixpkgs/stable/#chap-quick-start

File: pkgs/tools/system/swaysettings/default.nix [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+                  json-glib
+                  libgee
+                  pantheon.granite
+                  libxml2
+                  accountsservice
+                  gobject-introspection
+                  dbus
+                  ];
+
+  meta = with pkgs.lib; {
```
Comment (AndersonTorres) [inline pkgs/tools/system/swaysettings/default.nix:60]:
```suggestion
  meta = with lib; {
```

At least one year ago `lib` was split from `stdenv`

**Comment by AndersonTorres** [comment]
Not interested anymore, @aacebedo?

**Comment by aacebedo** [comment]
> Not interested anymore, @aacebedo?

Yes. I switched one my machine to nixos, applied the fixes and tested it this morning actually. Planning to deliver this weekend.

File: pkgs/tools/system/swaysettings/default.nix (hunk 1 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Ok corrected

File: pkgs/tools/system/swaysettings/default.nix (hunk 2 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Ok corrected

File: pkgs/tools/system/swaysettings/default.nix (hunk 3 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Ok corrected

File: pkgs/tools/system/swaysettings/default.nix (hunk 4 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Ok corrected

File: pkgs/tools/system/swaysettings/default.nix (hunk 5 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: No. My nixfile formater was not activated correctly. Corrected

File: pkgs/tools/system/swaysettings/default.nix (hunk 6 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Corrected with last version

File: pkgs/tools/system/swaysettings/default.nix (hunk 7 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Corrected

File: pkgs/tools/system/swaysettings/default.nix (hunk 8 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Corrected

File: pkgs/tools/system/swaysettings/default.nix (hunk 9 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Removed and put meta at the end

File: pkgs/top-level/all-packages.nix (hunk 10 above)
Comment (aacebedo) [reply pkgs/top-level/all-packages.nix:35439]: Ok moved

File: pkgs/tools/system/swaysettings/default.nix (hunk 11 above)
Comment (aacebedo) [reply pkgs/tools/system/swaysettings/default.nix]: Corrected

**Comment by aacebedo** [comment]
I did a fixup to see the difference from the original. When correction is confirm, I'll squash it and push a single commit.

**Comment by aacebedo** [comment]
I squashed the fixup. Is it ready for merge ?

File: maintainers/maintainer-list.nix [hunk 12]
```diff
@@ -119,6 +119,12 @@
     githubId = 12578560;
     name = "Quinn Bohner";
   };
+  aacebedo = {
```
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:122]: This file should be in a separate commit.

Also put this in alphabetical order.

File: pkgs/applications/misc/swaysettings/default.nix [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+, meson
+, ninja
+, pantheon
+, pkg-config
+, python3Full
+, stdenv
+, vala
+, wrapGAppsHook
+}:
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/applications/misc/swaysettings/default.nix:25]:
```suggestion

stdenv.mkDerivation rec {
```

File: pkgs/applications/misc/swaysettings/default.nix [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+, gtk3
+, json-glib
+, libgee
+, libhandy
+, libxml2
+, meson
+, ninja
+, pantheon
+, pkg-config
+, python3Full
```
Comment (AndersonTorres) [inline pkgs/applications/misc/swaysettings/default.nix:20]:
```suggestion

, python3
```

As I have said, pythonFull is not to be used here.

File: pkgs/applications/misc/swaysettings/default.nix [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+, wrapGAppsHook
+}:
+stdenv.mkDerivation rec {
+  pname = "swaysettings";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "ErikReider";
+    repo = "SwaySettings";
+    rev = "v0.3.0";
```
Comment (AndersonTorres) [inline pkgs/applications/misc/swaysettings/default.nix:32]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/misc/swaysettings/default.nix [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+  };
+
+  nativeBuildInputs = [
+    appstream-glib
+    desktop-file-utils
+    gettext
+    meson
+    ninja
+    pkg-config
+    python3Full
```
Comment (AndersonTorres) [inline pkgs/applications/misc/swaysettings/default.nix:43]:
```suggestion
    python3
```

The full Python is for interactive use primarily.

File: pkgs/applications/misc/swaysettings/default.nix [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+  '';
+
+  meta = with lib; {
+    description = "A GUI for configuring your sway desktop";
+    longDescription = ''
+      Sway settings enables easy configuration of a sway desktop environment
+      such as selection of application or icon themes.
+    '';
+    homepage = "https://github.com/ErikReider/SwaySettings";
+
```
Comment (AndersonTorres) [inline pkgs/applications/misc/swaysettings/default.nix:73]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 12 above)
Comment (aacebedo) [reply maintainers/maintainer-list.nix]: corrected

File: pkgs/applications/misc/swaysettings/default.nix (hunk 13 above)
Comment (aacebedo) [reply pkgs/applications/misc/swaysettings/default.nix:26]: corrected

File: pkgs/applications/misc/swaysettings/default.nix (hunk 14 above)
Comment (aacebedo) [reply pkgs/applications/misc/swaysettings/default.nix]: corrected

File: pkgs/applications/misc/swaysettings/default.nix (hunk 15 above)
Comment (aacebedo) [reply pkgs/applications/misc/swaysettings/default.nix]: corrected

File: pkgs/applications/misc/swaysettings/default.nix (hunk 16 above)
Comment (aacebedo) [reply pkgs/applications/misc/swaysettings/default.nix]: corrected

File: pkgs/applications/misc/swaysettings/default.nix (hunk 17 above)
Comment (aacebedo) [reply pkgs/applications/misc/swaysettings/default.nix]: corrected

**Comment by aacebedo** [comment]
rebased on upstream
